devices/Sen5x/Sen5xSensor.cs

```diff
@@ -0,0 +1,355 @@
+        }
+
+        /// <summary>
+        /// Writes the interval [s] of the periodic fan-cleaning. Please note that since this configuration is volatile, it will be reverted to the default value after a device reset.
+        /// </summary>
+        /// <param name="value">The new auto cleaning interval to be set.</param>
+        public void WriteAutoCleaningInterval(AutoCleaningIntervalParameters value)
+        {
+            Write(0x8004, TimeSpan.FromMilliseconds(20), value);
+        }
+
+        /// <summary>
+        /// This command returns the product name SEN5x (SEN50, SEN54 or SEN55).
+        /// </summary>
+        /// <returns>The product name SEN5x (SEN50, SEN54 or SEN55).</returns>
+        public string ReadProductName()
+        {
+            return ((StringValue)Read(0xd014, TimeSpan.FromMilliseconds(20), new StringValue(32))).Text;
+        }
+
+        /// <summary>
+        /// This command returns the requested serial number.
+        /// </summary>
+        /// <returns>The serial number.</returns>
+        public string ReadSerialNumber()
+        {
+            return ((StringValue)Read(0xd033, TimeSpan.FromMilliseconds(20), new StringValue(32))).Text;
+        }
+
+        /// <summary>
+        /// Get firmware version. There is no Major/Minor, only a single value.
+        /// </summary>
+        /// <returns>The firmware version.</returns>
+        public byte ReadFirmwareVersion()
+        {
+            var data = Read(0xd100, TimeSpan.FromMilliseconds(20), new byte[3]);
+            return data[0];
+        }
+
+        /// <summary>
+        /// Use this command to read the Device Status Register.
+        /// </summary>
+        /// <returns>The device status.</returns>
+        public DeviceStatus ReadDeviceStatus()
+        {
+            return (DeviceStatus)Read(0xd206, TimeSpan.FromMilliseconds(20), new DeviceStatus());
+        }
+
+        /// <summary>
+        /// Clears all flags in device status register.
+        /// </summary>
+        public void ClearDeviceStatus()
+        {
+            Read(0xd210, TimeSpan.FromMilliseconds(20));
+        }
+
+        /// <summary>
+        /// Device software reset command. After calling this command, the module is in the same state as after a power reset.
+        /// </summary>
+        public void DeviceReset()
+        {
+            Read(0xd304, TimeSpan.FromMilliseconds(100));
+        }
+
+        private AbstractReadEntity Read(ushort cmd, TimeSpan commandExecutionTime, AbstractReadEntity entity = null)
+        {
+            SpanByte data = new byte[2];
+            BinaryPrimitives.WriteUInt16BigEndian(data, cmd);
+            var result = _i2c.Write(data);
+            if (result.Status != I2cTransferStatus.FullTransfer)
+            {
+                throw new InvalidOperationException($"Unexpected status ({result.Status}) on I2C write");
```

Comment:
note that the exception pattern not to have the text apply here as well. Just extract it in the public functions calling it or just don't do it, there is only 2 functions where this expection is happening. So it's very easy to spot in the code as well.